docs: add documentation about lost reactivity from provide (#25054)
nuxt/nuxt: docs/2.guide/2.directory-structure/1.plugins.md

Note that we highly recommend using [`composables`](/docs/guide/directory-structure/composables) instead of providing helpers to avoid polluting the global namespace and keep your main bundle entry small.
::

::callout{color="amber" icon="i-ph-warning-duotone"}

> **If your plugin provides reactive properties (such as `ref`, `computed`, `reactive` and the likes) they will lose their template reactivity.** :br

Reviewer: ```suggestion
**If your plugin provides a `ref` or `computed`, they will not be unwrapped in a component `<template>`.** :br
```

> This is due to how Vue works with reactive objects that aren't top-level to the template. You can read more about it [here](https://vuejs.org/guide/essentials/reactivity-fundamentals.html#caveat-when-unwrapping-in-templates).

Reviewer 2: ```suggestion
This is due to how Vue works with reactive objects that aren't top-level to the template. You can read more about it [in the Vue documentation](https://vuejs.org/guide/essentials/reactivity-fundamentals.html#caveat-when-unwrapping-in-templates).
```

> This is due to how Vue works with reactive objects that aren't top-level to the template. You can read more about it [in the Vue documentation](https://vuejs.org/guide/essentials/reactivity-fundamentals.html#caveat-when-unwrapping-in-templates).

Reviewer: ```suggestion
This is due to how Vue works with refs that aren't top-level to the template. You can read more about it [in the Vue documentation](https://vuejs.org/guide/essentials/reactivity-fundamentals.html#caveat-when-unwrapping-in-templates).
```
